device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/plugandplay/PnpHelper.java

```diff
@@ -15,15 +15,16 @@
 
 import static java.util.Collections.singleton;
 import static java.util.Collections.singletonMap;
-
+/*
+ A helper class for formatting command requests and properties as per plug and play convention.
+*/
 public class PnpHelper {
 
     private static final String TELEMETRY_COMPONENT_NAME = "$.sub";
```

Comment:
Doesn't java have a property on Message class to set this directly?

also nit: I've update the .NET code to make all these public, and added a comment explaining what they mean and why they are required.